src/tracing_tool_output.mli

```diff
@@ -14,6 +24,9 @@ val param : t Command.Param.t
 val write_and_maybe_view
   :  ?num_temp_strs:int
   -> t
-  -> f_sexp:(Writer.t -> 'a Deferred.Or_error.t)
-  -> f_fuchsia:(Tracing_zero.Writer.t -> 'a Deferred.Or_error.t)
+  -> f:
+       (?events_writer:events_writer
```

Comment:
I think this parameter should be an `option`, not "optional". You have to pass a closure that takes this optional argument either way, might as well make the type reflect that fact.

And I think it's right in the sense that `f` really should be thinking about how to deal with no events writer explicitly.